lib/src/source/git.dart

```diff
@@ -284,9 +344,29 @@ class GitSource extends CachedSource {
   }
 
   /// Checks out the reference [ref] in [repoPath].
-  Future _checkOut(String repoPath, String ref) {
-    return git.run(["checkout", ref], workingDir: repoPath).then(
-        (result) => null);
+  Future _checkOut(String repoPath, String ref) async {
```

Comment:
Note that you won't need to modify this if you change `downloadToSystemCache` to use `_getRef`.